### ISSUE TITLE
Missing function: isEmpty

### ISSUE THREAD

**Issue by paour** [remeda/remeda#307]
I've seen a few references to this function in issues, but it's neither implemented, nor documented in `mapping.md` and I was wondering why.

It's all the more important because checking for empty objects (in particular) is a fraught subject in JS. https://www.samanthaming.com/tidbits/94-how-to-check-if-object-is-empty/

**Comment by TkDodo** [comment]
I don't think with TypeScript, you would need the constructor check. An `isEmpty` function for objects should only accept objects, so the keys-length-check should suffice, no? We could add that to the one-liner helpers?

**Comment by paour** [comment]
Array and string overloads would be nice.

**Comment by eranhirsch** [comment]
I commented on the PR but just mirroring what I said there:
1. With stronger typing overall in Remeda the value of these predicates is in narrowing types, so you'd want a isEmpty(x) to narrow the *type* of `x` to it's trivial empty value.
2. But narrowing an array to [], or a string to "", isn't very useful as you most likely not act on the empty array itself (you'd probably have some fallback logic instead).
3. This means that most likely what you actually want is a `isNonEmpty`, as that would ensure that, for example, `myArray[0]` is defined.
4. The problem is that typescript doesn't do implicit typing for *negated type predicates*, so simply doing `!isEmpty(x)` is meaningless.
5. And defining isNonEmpty on anything *but arrays* is impossible in typescript as there's no way to type a non-empty string, or an object with at least one prop in it (and it's especially hard if you actually want to refine the input, and not just return a generic non-empty value).

TL;DR - I don't think this is very useful as a type predicate, and as a simple boolean predicate this is just a trivial one liner.

**Comment by paour** [comment]
Fixed by #314
